src/dmd/astbase.d

```diff
@@ -128,6 +128,7 @@ struct ASTBase
         scopeinferred       = (1L << 49),   // 'scope' has been inferred and should not be part of mangling
         future              = (1L << 50),   // introducing new base class function
         local               = (1L << 51),   // do not forward (see dmd.dsymbol.ForwardingScopeDsymbol).
+        returninferred      = (1L << 52),   // 'return' has been inferred and should not be part of mangling
```

Comment:
Wouldn't it be better to have just one STCinferred?

Reply:
Either return or scope or both can be present or inferred.

Reply:
I'd not have expected inference to happen if either is explicitly given, but Ok.

Some of the bit flags may require a little pruning as there aren't many available left now.